crypto/evp/evp_pbe.c

```diff
@@ -258,9 +288,13 @@ int EVP_PBE_find_ex(int type, int pbe_nid, int *pcnid, int *pmnid,
     pbelu.pbe_type = type;
     pbelu.pbe_nid = pbe_nid;
 
-    if (pbe_algs != NULL) {
+    if (ensure_init()) {
+        if (!CRYPTO_THREAD_write_lock(pbe_algs_lock))
```

Comment:
Maybe a comment about the later find requiring a write lock would be appropriate?